### PR TITLE
Update angular to Angular 4.0

### PR DESCRIPTION
**Pull request by SpencerCornish**
Update to the new angular, and update basically everything else that had to be!  

Added Packages:
`angular_router:^1.0.2`
`angular_forms: "^1.0.0"`

Updated Packages:
- angular2: `^3.0.0` to `^4.0.0`
- build: `^0.7.0` to  `^0.10.2+1`
- build_runner: `^0.3.0` to `^0.5.0`
- built_value_generator: `^1.0.0` to `^4.2.0`

#### Acceptance Criteria:
  Still runs!

### CODE REVIEW

**Comment by jerold** [comment]
:O

**Comment by jerold** [comment]
@SpencerCornish I doubt this broke in this PR, but I noticed shred isn't working. That's the only feature that wasn't 100% though. Checking out the code changes now...

**Review by jerold — APPROVED** [review]
Looks like the majority of this PR is dartfmt changes. But otherwise the conversion changes look good. Thanks for doing this!

**Comment by jeroldalbertson-wf** [comment]
@SpencerCornish there is a part of [_shredSession missing](https://github.com/davidmarne/retro/blob/master/lib/src/middleware/refMiddleware.dart#L301). I don't think the logic in master is really that good. since we should make sure board is good before shredding it.

File: lib/src/middleware/refMiddleware.dart [hunk 1]
```diff
@@ -148,184 +146,168 @@ _updateCurrentSessionSubs(FirebaseClient client, MiddlewareApi<App, AppBuilder,
...
+_addSupport(FirebaseClient client) =>
+    (MiddlewareApi<App, AppBuilder, AppActions> api, ActionHandler next,
+        Action<String> action) {
+      next(action);
+      var userUid = api.state.users.currentUid;
+      Item item = api.state.items.map[action.payload];
+      if (item != null && userUid != "") {
+        client.addSupport(userUid, item);
       }
-      client.shredSession(session, board);
```
Comment (jeroldalbertson-wf) [inline lib/src/middleware/refMiddleware.dart]: This is important. :P

**Comment by jeroldalbertson-wf** [comment]
If [this pr](https://github.com/davidmarne/retro/pull/42) looks good please feel free to merge it into your branch, then I can merge this. :)

**Comment by jerold** [comment]
@SpencerCornish is this ready to merge? :) If you're still adding stuff I'll hold off.

**Comment by SpencerCornish** [comment]
@jerold It's ready!

**Pull request merged**